test/jasmine/tests/hover_label_test.js

```diff
@@ -0,0 +1,245 @@
+var d3 = require('d3');
+
+var Plotly = require('@src/index');
+var Fx = require('@src/plots/cartesian/graph_interact');
+var Lib = require('@src/lib');
+
+var createGraphDiv = require('../assets/create_graph_div');
+var destroyGraphDiv = require('../assets/destroy_graph_div');
+
+describe('hover info', function() {
+    'use strict';
+
+    var mock = require('@mocks/14.json'),
+        evt = {
+                clientX: mock.layout.width/ 2,
+                clientY: mock.layout.height / 2
+        };
+
+    afterEach(destroyGraphDiv);
+
+    describe('hover info', function() {
+        var mockCopy = Lib.extendDeep({}, mock);
+
+        beforeEach(function(done) {
+            Plotly.plot(createGraphDiv(), mockCopy.data, mockCopy.layout).then(done);
```

Comment:
ah aight - the only other suggestion the to remove the multiple BeforeEach and Describes is to do something like

```javascript

it('does hover things', function(done) {
    Plotly.plot(newGD(), newData(), newLayout()).then( function () {
       expect(thingsToHappen()).toBe(true)
       done()
    })
})
```

where you get shorter code but have 1 level of nesting in the tests.